### PR TITLE
Added handling of network state request

### PR DESCRIPTION
**Pull request by axlan**
Depends on https://github.com/swift-nav/libsbp/pull/449

### CODE REVIEW

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-66-g377c2b1
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-66-g377c2b1/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-66-g377c2b1/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-66-g377c2b1

File: package/piksi_system_daemon/src/main.c [hunk 1]
```diff
@@ -384,6 +393,85 @@ static int command_output(zloop_t *loop, zmq_pollitem_t *item, void *arg)
...
+        }
+      }
+      if (interface_idx == SBP_MAX_NETWORK_INTERFACES) {
+        perror("Too many interfaces");
+        continue;
+      }
+
+      family = ifa->ifa_addr->sa_family;
+      if (family == AF_INET) {
+        interfaces[interface_idx].ip_address = endian_flip_u32(((struct sockaddr_in *)ifa->ifa_addr)->sin_addr.s_addr);
```
Comment (jacobmcnamee) [inline package/piksi_system_daemon/src/main.c]: There are standard functions for this type of conversion, e.g. `htons()` and `inet_ntoa()`

File: package/piksi_system_daemon/src/main.c [hunk 2]
```diff
@@ -384,6 +393,85 @@ static int command_output(zloop_t *loop, zmq_pollitem_t *item, void *arg)
...
+      if (ifa->ifa_addr == NULL)
+          continue;
+
+      if (strlen(ifa->ifa_name) >= sizeof(interfaces[0].interface_name)) {
+        perror("Interface name too long");
+        continue;
+      }
+
+      int interface_idx;
+      for (interface_idx = 0; interface_idx < SBP_MAX_NETWORK_INTERFACES; interface_idx++)
```
Comment (jacobmcnamee) [inline package/piksi_system_daemon/src/main.c]: Why do you need a nested loop here? Can you simply do
`strcpy(interfaces[n].interface_name, ifa->ifa_name);`
where `n` is the (looks like currently unused) index variable in the outer loop?

Comment (jacobmcnamee) [inline package/piksi_system_daemon/src/main.c]: Or alternatively just use a separate variable initialized outside the outer loop.

Comment (axlan) [inline package/piksi_system_daemon/src/main.c]: Interfaces appear multiple times (with IPv4, IPv6, and stats). This checks to see if this interface already appeared and if so add to the existing entry.

**Comment by axlan** [comment]
Tested working in hardware

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-83-gcd92cdc
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-83-gcd92cdc/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-83-gcd92cdc/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-83-gcd92cdc

**Comment by anonymous-travis-swiftnav** [comment]
## v1.0.0-branch-83-gcf67ccb
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-83-gcf67ccb/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.0.0-branch-83-gcf67ccb/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.0.0-branch-83-gcf67ccb

**Comment by anonymous-travis-swiftnav** [comment]
## v1.1.0-branch-89-g837769c
+ [Artifacts (HITL Dashboard Page)](http://hitl-dashboard.swiftnav.com/files/swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-89-g837769c/)
+ [Artifacts (S3 / AWS Console)](https://console.aws.amazon.com/s3/home?region=us-west-2&bucket=swiftnav-artifacts-pull-requests&prefix=piksi_buildroot/v1.1.0-branch-89-g837769c/)
+ s3://swiftnav-artifacts-pull-requests/piksi_buildroot/v1.1.0-branch-89-g837769c

**Pull request merged**